### PR TITLE
38 nodeguard withdrawals only allow to add negative btc amount

### PR DESCRIPTION
**Pull request by markettes**
Done:
- Name changing for the withdrawal max env var
- Logic changed for handling env var
- Min and max set to numeric field

### CODE REVIEW

File: src/Helpers/ValidationHelper.cs [hunk 1]
```diff
@@ -58,14 +58,22 @@ public static void ValidateWithdrawalAmount(ValidatorEventArgs obj, Boolean isAm
 
         obj.Status = ValidationStatus.Success;
 
-        var environmentVariable = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? throw new InvalidOperationException();
-
-        var minimum = decimal.Parse(environmentVariable, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture);
+        var environmentVariableMin = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? throw new InvalidOperationException();
```
Comment (Jossec101) [inline src/Helpers/ValidationHelper.cs:61]: I think the minimum should be 0 if the env-var is not set

Comment (Jossec101) [reply src/Helpers/ValidationHelper.cs:61]:
```suggestion
        var environmentVariableMin = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? 0.0M;
```

**Review by Jossec101 — APPROVED** [review]

File: src/Helpers/ValidationHelper.cs [hunk 2]
```diff
@@ -58,14 +58,22 @@ public static void ValidateWithdrawalAmount(ValidatorEventArgs obj, Boolean isAm
 
         obj.Status = ValidationStatus.Success;
 
-        var environmentVariable = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? throw new InvalidOperationException();
-
-        var minimum = decimal.Parse(environmentVariable, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture);
+        var environmentVariableMin = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? 0.0M;
```
Comment (Jossec101) [inline src/Helpers/ValidationHelper.cs:61]:
```suggestion
        var environmentVariableMin = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? 0.0m;
```

**Review by Jossec101 — DISMISSED** [review]

File: src/Helpers/ValidationHelper.cs [hunk 3]
```diff
@@ -58,14 +58,22 @@ public static void ValidateWithdrawalAmount(ValidatorEventArgs obj, Boolean isAm
 
         obj.Status = ValidationStatus.Success;
 
-        var environmentVariable = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? throw new InvalidOperationException();
-
-        var minimum = decimal.Parse(environmentVariable, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture);
+        var environmentVariableMin = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT") ?? 0.0m;
+        var environmentVariableMax = Environment.GetEnvironmentVariable("MAXIMUM_WITHDRAWAL_BTC_AMOUNT") ?? throw new InvalidOperationException();
```
Comment (Jossec101) [inline src/Helpers/ValidationHelper.cs:62]:
```suggestion
        var environmentVariableMax = Environment.GetEnvironmentVariable("MAXIMUM_WITHDRAWAL_BTC_AMOUNT") ?? 21_000_000m;
```

**Review by Jossec101 — APPROVED** [review]

**Review by Jossec101 — APPROVED** [review]

File: src/Helpers/ValidationHelper.cs (hunk 3 above)
Comment (markettes) [reply src/Helpers/ValidationHelper.cs]: Yes, that looks better

Comment (markettes) [reply src/Helpers/ValidationHelper.cs]: This doesnt compiles

Comment (markettes) [reply src/Helpers/ValidationHelper.cs]:
```
var environmentVariableMin = Environment.GetEnvironmentVariable("MINIMUM_WITHDRAWAL_BTC_AMOUNT");
var environmentVariableMax = Environment.GetEnvironmentVariable("MAXIMUM_WITHDRAWAL_BTC_AMOUNT");
decimal minimum, maximum;
if (environmentVariableMin == null) minimum = 0.0m;
else minimum = decimal.Parse(environmentVariableMin, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture);
if (environmentVariableMax == null) maximum = 21_000_000;
else maximum = decimal.Parse(environmentVariableMax, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture);
```

This should work

**Pull request merged**